lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/ProcessJourneyEventHandler.java

```diff
@@ -253,41 +252,51 @@ private StepResponse executeJourneyEvent(
                             .with(LOG_JOURNEY_EVENT.getFieldName(), journeyEvent)
                             .with(
                                     LOG_JOURNEY_TYPE.getFieldName(),
-                                    ipvSessionItem.getJourneyType()));
-            throw new JourneyEngineException();
-        } catch (NoCurrentStateException e) {
-            LOGGER.error(
-                    LogHelper.buildErrorMessage("No current state found for user in stack", e)
-                            .with(LOG_JOURNEY_EVENT.getFieldName(), journeyEvent));
+                                    ipvSessionItem.getState().subJourney().name()));
             throw new JourneyEngineException();
         }
     }
 
     @Tracing
     private State executeStateTransition(
+            JourneyState initialJourneyState,
             IpvSessionItem ipvSessionItem,
             String journeyEvent,
             String currentPage,
             AuditEventUser auditEventUser,
             String deviceInformation)
             throws StateMachineNotFoundException, SqsException, UnknownEventException,
                     UnknownStateException {
-        StateMachine stateMachine = stateMachines.get(ipvSessionItem.getJourneyType());
+
+        StateMachine stateMachine = stateMachines.get(initialJourneyState.subJourney());
         if (stateMachine == null) {
             throw new StateMachineNotFoundException(
                     String.format(
                             "State machine not found for journey type: '%s'",
-                            ipvSessionItem.getJourneyType()));
+                            initialJourneyState.subJourney()));
         }
         LOGGER.debug(
                 LogHelper.buildLogMessage(
                         String.format(
                                 "Found state machine for journey type: %s",
-                                ipvSessionItem.getJourneyType().name())));
+                                initialJourneyState.subJourney())));
+
+        if (BACK_EVENT.equals(journeyEvent) && !isBackEventDefinedOnState(initialJourneyState)) {
+            var previousJourneyState = ipvSessionItem.getPreviousState();
+
+            if (isPageState(initialJourneyState) && isPageState(previousJourneyState)) {
+                ipvSessionItem.pushState(previousJourneyState);
```

Comment:
I guess I was thinking that the user is only going to go 'back' once, and this would give us a fuller picture of there journey in the stack. But you're right, any more than once and it'll break. I'll switch it, it's probably not that important to have that information in the stack anyway.